### ISSUE TITLE
Web Host Messaging: Unified Inbox for Guest Communication (Email, Text, Whatsapp)

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#767]
## 🚀 Feature Summary

The Unified Inbox brings all guest, host and team communications from multiple channels into one centralized view. It enables teams to respond faster, collaborate efficiently, and maintain consistent messaging across platforms. Automated messages, templates, and conversation history help deliver a seamless guest experience. Whether a guest is messaging you on WhatsApp, Email, SMS, OTA, or through a direct booking site, you can receive and respond from the Domits inbox. 

----

[PMS](https://github.com/domits1/Domits/issues/149) Capabilities: [Listings](https://github.com/domits1/Domits/issues/2792), [CRS](https://github.com/domits1/Domits/issues/501), [CM](https://github.com/domits1/Domits/issues/440), [Direct Bookings](https://github.com/domits1/Domits/issues/2600), [RMS](https://github.com/domits1/Domits/issues/436), [Calendar & Prices](https://github.com/domits1/Domits/issues/36), [Inbox](https://github.com/domits1/Domits/issues/767), [Tasks](https://github.com/domits1/Domits/issues/2794), [Finance](https://github.com/domits1/Domits/issues/163), [Reviews](https://github.com/domits1/Domits/issues/90), [Intelligence](https://github.com/domits1/Domits/issues/1490), [Settings](https://github.com/domits1/Domits/issues/2793), [Integrations](https://github.com/domits1/Domits/issues/433), [Agentic AI](https://github.com/domits1/Domits/issues/442)
Use Case / Scenario Testing: [PMS](https://github.com/domits1/Domits/issues/2262), [CRS](https://github.com/domits1/Domits/issues/2265), [CM](https://github.com/domits1/Domits/issues/2266), [RMS](https://github.com/domits1/Domits/issues/2267), [Calendar & Prices](https://github.com/domits1/Domits/issues/2299), [Inbox](https://github.com/domits1/Domits/issues/2291), [Finance](https://github.com/domits1/Domits/issues/2287)

----

<img width="1306" height="517" alt="Image" src="https://github.com/user-attachments/assets/9ba456e1-7e5e-43bd-8cf6-9dac9d21e8ba" />

----
### Main messaging experience 
- #89 

### General 'internal domits' guest communication
- [x] #536 
- [x] #2826 

### Unified inbox preferences
- [ ] #2260
- [ ] #2777 

### Automated inbox 
- [ ] #768 
- [ ] #2777 

### External messaging integration (OTAs's etc)
- [ ] #2288
- [ ] #2616
- [ ] Web Host Messaging: Unified Inbox: VRBO (Expedia) Integration Overview, Management & API Requirements

### External messaging integration (standardized communication tools ex: email, sms)
- [ ] #1251
- [ ] #2615
- [ ] #2614
- [ ] Web Host Messaging: Unified Inbox: Instagram, Facebook

- [ ] More
- [ ] Also connected with Discord/Slack, you have the option to have Domits fit in with your centralized channel manager, further simplifying channel management. For instance, when a guest sends an urgent message, Domits notifies the host or property manager through Discord/Slack and allows them to respond. 

A Unified Messaging Inbox consolidates all messages from guests, hosts, cleaning staff, and OTAs (Airbnb, Booking.com, VRBO, direct bookings, etc.) into a single thread per reservation or contact — streamlining communication across platforms.

---

## 🧩 Problem Statement
What problem does this feature solve? Who is affected, and why does it matter? > Focus on the problem, not the solution.

No centralized unified inbox

---

## 🎯 Desired Outcome
What does success look like if this feature is implemented?

Centralized unified inbox

---

## 👥 Target Users
Who benefits from this feature?
- [x] Guests / Customers
- [x] Hosts / Property Managers / Enterprise
- [ ] Admin / Internal users
- [ ] Developers
- [ ] Other: _______

---

## 📈 Business & User Impact
How valuable is this feature?

- **Impact level:**
  - [ ] Low (nice-to-have)
  - [ ] Medium (improves workflows)
  - [ ] High (unblocks key use cases)
  - [x] Critical (strategic / revenue-impacting)

- **User reach:**
  - [ ] Few users
  - [ ] Many users
  - [x] All users
  - [ ] Unknown

---

## ⚠️ Constraints & Considerations
- Performance
- Security
- Privacy
- Accessibility
- Backward compatibility
- External dependencies

---

## 🧪 Acceptance Criteria / User Stories
How can we validate that this feature is complete?

- [ ] Given X, when Y, then Z

👥 User Roles

- Guest
- Host / Property Manager
- Team Member (Cleaner, Maintenance, etc.)
- Admin / Supervisor
- Integration (OTA / Channel API)

💬 1. Messaging Basics

- [ ] As a host, I want to send and receive messages from all channels in one inbox.
- [ ] As a host, I want to reply to messages from Airbnb, Booking.com, and direct channels without switching tabs.
- [ ] As a guest, I want to communicate with my host through the booking platform securely.
- [ ] As a system, I want to automatically match messages to the correct reservation.
- [ ] As a host, I want to compose and send new messages manually.
- [ ] As a host, I want to attach files and images to messages.
- [ ] As a guest, I want to receive automatic responses when hosts are unavailable.
- [ ] As a host, I want to mark a conversation as read or unread.
- [ ] As a host, I want to archive old conversations.
- [ ] As a system, I want to display timestamps and message status (sent, delivered, read).

📬 2. Multi-Channel Integration

- [ ] As a system, I want to import messages from Airbnb API.
- [ ] As a system, I want to import messages from Booking.com XML API.
- [ ] As a system, I want to import messages from VRBO and Expedia channels.
- [ ] As a system, I want to sync messages sent through the PMS booking engine.
- [ ] As an admin, I want to configure which channels are connected to the unified inbox.
- [ ] As a host, I want to see the source channel (e.g., Airbnb, Direct, Booking.com) on each thread.
- [ ] As a system, I want to normalize different message formats from APIs into a single view.
- [ ] As a system, I want to automatically retry message delivery if a channel API fails temporarily.
- [ ] As an admin, I want to view API sync logs for troubleshooting.

📂 3. Conversation Organization

- [ ] As a host, I want each guest or reservation to have a single conversation thread.
- [ ] As a host, I want to filter conversations by property, date range, or channel.
- [ ] As a host, I want to tag messages with labels (urgent, pending payment, housekeeping).
- [ ] As a host, I want to assign conversations to team members.
- [ ] As a supervisor, I want to monitor who’s assigned to each thread.
- [ ] As a host, I want to flag conversations for follow-up.
- [ ] As a host, I want to view unread message count per channel.
- [ ] As a system, I want to auto-assign messages based on property ownership.
- [ ] As a host, I want to merge duplicate threads if they relate to the same guest.
- [ ] As an admin, I want to set default filters (e.g., “Unanswered first”).

🤖 4. Automations & Templates

- [ ] As a host, I want to send automated welcome messages after booking confirmation.
- [ ] As a host, I want to schedule a check-in instructions message 24 hours before arrival.
- [ ] As a host, I want to create message templates with placeholders (guest name, property name, check-in time).
- [ ] As a system, I want to auto-send thank-you messages after checkout.
- [ ] As a host, I want to define trigger rules for automation (e.g., after booking, before check-in).
- [ ] As an admin, I want to manage shared message templates across multiple users.
- [ ] As a host, I want to pause automations for a specific reservation.
- [ ] As a system, I want to prevent duplicate automation messages.
- [ ] As a host, I want to test automated messages before activating them.
- [ ] As a system, I want to log all automated communications for auditing.

🧑‍💼 5. Internal Team Messaging

- [ ] As a host, I want to send internal notes within a conversation (not visible to the guest).
- [ ] As a cleaner, I want to receive cleaning task messages through the same inbox.
- [ ] As a maintenance worker, I want to update task completion through internal chat.
- [ ] As a supervisor, I want to tag staff in messages using @mentions.
- [ ] As a host, I want to differentiate between guest and team messages visually.
- [ ] As an admin, I want to control which roles can access guest conversations.
- [ ] As a system, I want to sync task updates (cleaning, maintenance) with the operations calendar.

🔔 6. Notifications & Alerts

- [ ] As a host, I want to receive push notifications for new messages.
- [ ] As a host, I want to receive desktop and email alerts for unread messages.
- [ ] As a system, I want to batch notifications to avoid overload.
- [ ] As a host, I want to snooze notifications for a certain period.
- [ ] As an admin, I want to control global notification preferences.
- [ ] As a host, I want to view notification history for missed messages.
- [ ] As a system, I want to detect urgent messages (e.g., “locked out”) and flag them.
- [ ] As a host, I want to mark notifications as resolved once a message is read.

📊 7. Reporting & Analytics

- [ ] As a host, I want to view response time metrics for my team.
- [ ] As a supervisor, I want reports showing message volume per property or channel.
- [ ] As an admin, I want to measure first-response time SLA.
- [ ] As a host, I want to see which automations have the highest engagement rate.
- [ ] As a manager, I want to export communication reports in CSV format.
- [ ] As a host, I want to analyze frequent guest questions for improving templates.
- [ ] As an admin, I want to identify peak messaging hours.
- [ ] As a supervisor, I want to track performance by team member (messages handled, response speed).

🔍 8. Search & Filters

- [ ] As a host, I want to search messages by guest name, reservation ID, or keyword.
- [ ] As a host, I want to filter by unread, flagged, or assigned messages.
- [ ] As a host, I want to filter by date range.
- [ ] As an admin, I want to view only system or automation messages.
- [ ] As a host, I want to quickly find conversations from specific OTAs.
- [ ] As a system, I want to cache recent searches for faster access.
- [ ] As a host, I want to save custom filter views (e.g., “My guests this week”).

🧾 9. Reservation & Guest Context Integration

- [ ] As a host, I want to view booking details (check-in, checkout, property) inside the message thread.
- [ ] As a host, I want to see guest profile information while messaging.
- [ ] As a system, I want to auto-link conversations to the correct reservation ID.
- [ ] As a host, I want to update reservation status (confirmed, checked-in) from the inbox.
- [ ] As a host, I want to open the reservation directly from the message view.
- [ ] As a host, I want to see payment and balance information within the chat.

🧠 10. AI & Smart Assistance (optional advanced module)

- [ ] As a system, I want to auto-suggest message replies using AI based on guest intent.
- [ ] As a host, I want AI to summarize long message threads.
- [ ] As a host, I want the system to detect sentiment in guest messages (positive, neutral, negative).
- [ ] As a host, I want the system to automatically recommend relevant templates.
- [ ] As an admin, I want to enable or disable AI features per property.

🌍 11. Localization & Multi-Language Support

- [ ] As a system, I want to auto-translate messages between host and guest languages.
- [ ] As a host, I want to see both original and translated messages.
- [ ] As a system, I want to detect guest language automatically.
- [ ] As a host, I want to choose default reply language.
- [ ] As an admin, I want to manage supported translation APIs.

🔐 12. Security & Compliance

- [ ] As a system, I want to mask sensitive data (credit cards, phone numbers) in messages.
- [ ] As an admin, I want to restrict user access by role and property.
- [ ] As a system, I want all communication stored securely and encrypted.
- [ ] As an admin, I want audit logs for message edits, sends, and deletions.
- [ ] As a system, I want to comply with GDPR data retention rules.
- [ ] As a host, I want to delete guest messages upon request.

🧱 13. Scalability & Performance

- [ ] As a system, I want to handle large volumes of concurrent messages.
- [ ] As an admin, I want to monitor system uptime and API latency.
- [ ] As a system, I want to queue messages during OTA outages.
- [ ] As an admin, I want performance dashboards for message throughput.
- [ ] As a system, I want to automatically retry failed sends.

🧩 14. Integrations

- [ ] As an admin, I want to integrate the inbox with PMS and CRS systems.
- [ ] As a host, I want to link guest conversations to CRM profiles.
- [ ] As an admin, I want webhook support to push message events to third-party tools (Slack, email, Zapier).
- [ ] As a host, I want to respond to messages directly from my mobile app.
- [ ] As an admin, I want API endpoints for developers to access message data securely.

**Comment by Timmhart** [comment]
See #977
